src/tests/integration/RateSourceService.test.ts

```diff
@@ -0,0 +1,80 @@
+/* eslint-disable */
+import request from 'supertest';
+import * as sinon from 'sinon';
+import { app } from '../../app';
+
+describe('RateSourceService', () => {
+  let fetchStub: sinon.SinonStub;
+
+  beforeEach(() => {
+    fetchStub = sinon.stub(global, 'fetch');
+  });
+
+  afterEach(sinon.restore);
+
+  test('should respond with current rate and 200 code', async () => {
+    const ratesData = {
+      json: () => {
+        return Promise.resolve({
+          meta: {
+            code: 200,
+          },
+          rates: {
+            UAH: 40.33,
+          },
+        });
+      },
+    };
+    fetchStub.resolves(ratesData);
+
+    const response = await request(app).get('/rate');
+
+    expect(response.status).toBe(200);
+    expect(response.body).toStrictEqual({ rate: 40.33 });
+  });
+
+  test('should respond 429 error', async () => {
+    const errorResponseFromThirdPartyService = {
+      meta: {
+        code: 429,
+        error_type: 'auth failed',
+        error_detail: 'Request limits exceeded',
+      },
+    };
+    const jsonErrorResponseFromThirdPartyService = {
+      json: () => {
+        return Promise.resolve(errorResponseFromThirdPartyService);
+      },
+    };
+    fetchStub.resolves(jsonErrorResponseFromThirdPartyService);
+
+    const response = await request(app).get('/rate');
+    expect(response.status).toBe(429);
+    expect(response.body).toStrictEqual({
+      error: `${errorResponseFromThirdPartyService.meta.error_type}: ${errorResponseFromThirdPartyService.meta.error_detail}`,
+    });
+  });
```

Comment:
not sure if this tests do tests anything since the response is mocked

Reply:
Actually this is the test that found a huge problem within my app, because I previously didn't handle errors form third-party service properly and it crashed my app

So now I'm passing status codes and errors to my endpoints

And when I add another third party services - I need these tests to work, I hope that makes sense :)